pkg/volume/azure_file/azure_provision.go

```diff
@@ -63,12 +63,17 @@ func (plugin *azureFilePlugin) newDeleterInternal(spec *volume.Spec, util azureU
 	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.AzureFile == nil {
 		return nil, fmt.Errorf("invalid PV spec")
 	}
+	nameSpace := ""
 	pvSpec := spec.PersistentVolume
-	if pvSpec.Spec.ClaimRef.Namespace == "" {
-		glog.Errorf("namespace cannot be nil")
-		return nil, fmt.Errorf("invalid PV spec: nil namespace")
+	if pvSpec.Spec.AzureFile.SecretNameSpace != nil {
```

Comment:
this could be ""... if non-nil, I would assign, then if namespace is still empty, pull from the claimRef.Namespace